shared/validate/validate.go

```diff
@@ -154,6 +154,26 @@ func IsDeviceID(value string) error {
 	return nil
 }
 
+// IsInterfaceName validates a real network interface name.
+func IsInterfaceName(value string) error {
```

Comment:
@brauner we had this function (albeit it under a different name) already, so I have used it to validate the NIC's interface `name` property, which as well as preventing `%` being passed to liblxc/kernel will also avoid using too short/long interface names or any with invalid characters.

Reply:
Ah, excellent!